<a href="https://colab.research.google.com/github/usshaa/ASAP/blob/main/Salary_Analysis_of_San_Francisco_Employees.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Project: Salary Analysis of San Francisco Employees**
### **Project Questions:**
1. **What is the average (mean) base salary of employees?**
2. **Which job title has the highest median base salary?**
3. **What is the most frequently occurring base salary (mode)?**

In [60]:
# Importing required libraries
import pandas as pd
import statistics as stats
import matplotlib.pyplot as plt
import seaborn as sns

In [40]:
#!/bin/bash
!kaggle datasets download kaggle/sf-salaries

Dataset URL: https://www.kaggle.com/datasets/kaggle/sf-salaries
License(s): CC0-1.0
sf-salaries.zip: Skipping, found more recently modified local copy (use --force to force download)


In [42]:
!unzip sf-salaries.zip


Archive:  sf-salaries.zip
replace Salaries.csv? [y]es, [n]o, [A]ll, [N]one, [r]ename: y
  inflating: Salaries.csv            
replace database.sqlite? [y]es, [n]o, [A]ll, [N]one, [r]ename: y
  inflating: database.sqlite         


In [43]:
# Load the dataset
df = pd.read_csv("Salaries.csv")

# Display first few rows
df.head()

<ipython-input-43-e53f024db686>:2: DtypeWarning: Columns (3,4,5,6,12) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("Salaries.csv")


Id       EmployeeName                                        JobTitle  \
0   1     NATHANIEL FORD  GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY   
1   2       GARY JIMENEZ                 CAPTAIN III (POLICE DEPARTMENT)   
2   3     ALBERT PARDINI                 CAPTAIN III (POLICE DEPARTMENT)   
3   4  CHRISTOPHER CHONG            WIRE ROPE CABLE MAINTENANCE MECHANIC   
4   5    PATRICK GARDNER    DEPUTY CHIEF OF DEPARTMENT,(FIRE DEPARTMENT)   

     BasePay OvertimePay   OtherPay Benefits   TotalPay  TotalPayBenefits  \
0  167411.18         0.0  400184.25      NaN  567595.43         567595.43   
1  155966.02   245131.88  137811.38      NaN  538909.28         538909.28   
2  212739.13   106088.18    16452.6      NaN  335279.91         335279.91   
3    77916.0    56120.71   198306.9      NaN  332343.61         332343.61   
4   134401.6      9737.0  182234.59      NaN  326373.19         326373.19   

   Year  Notes         Agency Status  
0  2011    NaN  San Francisco    NaN  
1  2011    NaN  San Francisco    NaN  
2  2011    NaN  San Francisco    NaN  
3  2011    NaN  San Francisco    NaN  
4  2011    NaN  San Francisco    NaN

### **1. Compute Mean (Average) Base Salary**

In [44]:
# Convert BasePay to numeric and drop NaN values
df['BasePay'] = pd.to_numeric(df['BasePay'], errors='coerce')
df = df.dropna(subset=['BasePay'])

In [45]:
# Calculate mean salary
mean_salary = stats.mean(df['BasePay'])
print(f"Mean (Average) Base Salary: ${mean_salary:.2f}")

Mean (Average) Base Salary: $66325.45


✅ **Answer:** The mean salary is **$66325.45** (Example output; will change based on dataset).

### **2. Find the Job Title with the Highest Median Base Salary**

In [46]:
# Group by JobTitle and find median salary
job_median_salary = df.groupby('JobTitle')['BasePay'].median().sort_values(ascending=False)
print(job_median_salary.head(10))  # Display top 10 jobs with highest median salaries

JobTitle
Chief of Police                           307450.040
Chief, Fire Department                    302068.000
Gen Mgr, Public Trnsp Dept                294000.180
CHIEF OF DEPARTMENT, (FIRE DEPARTMENT)    285262.000
Mayor                                     281537.170
Dep Dir for Investments, Ret              276153.765
Dept Head V                               270616.270
Adm, SFGH Medical Center                  268946.020
Controller                                267914.000
EXECUTIVE CONTRACT EMPLOYEE               264452.520
Name: BasePay, dtype: float64


In [47]:
# Identify the job with the highest median salary
highest_median_job = job_median_salary.idxmax()
highest_median_value = job_median_salary.max()

In [56]:
print(f"Job with Highest Median Salary: {highest_median_job} (${highest_median_value:.2f})")

Job with Highest Median Salary: Chief of Police ($307450.04)


✅ **Answer:** The **Chief of Police** has the highest median salary of **$307450.04**.

### **3. Find the Most Common Salary (Mode)**

In [57]:
# Remove rows where BasePay is 0 or NaN
filtered_df = df[df['BasePay'] > 0]

# Calculate mode again
mode_salary = stats.mode(filtered_df['BasePay'])
print(f"Most Common Salary (Mode): ${mode_salary:.2f}")

Most Common Salary (Mode): $54703.00


✅ **Answer:** The most frequently occurring salary is **$54703.00**.


### **4. Impact of Overtime Pay on Total Salary**

## **Key Takeaways**
📌 **Mean vs. Median:**  
- If the mean salary is **higher than the median**, it indicates a **few very high salaries are pulling up the average**.
- If they are **similar**, salaries are more **evenly distributed**.

📌 **Mode Salary Insights:**  
- The **most common salary** helps **HR decide salary benchmarks** for new hires.